crates/bevy_transform/src/components/transform.rs

```diff
@@ -1,36 +1,126 @@
-use bevy_math::Mat4;
+use bevy_math::{Mat4, Quat, Vec3};
 use bevy_property::Properties;
 use std::fmt;
 
 #[derive(Debug, PartialEq, Clone, Copy, Properties)]
 pub struct Transform {
-    pub value: Mat4,
-    pub sync: bool, // NOTE: this is hopefully a temporary measure to allow setting the transform directly.
-                    // ideally setting the transform automatically propagates back to position / translation / rotation,
-                    // but right now they are always considered the source of truth
+    local: Mat4,
+    global: Mat4,
 }
 
 impl Transform {
```

Comment:
I like this api in general, but I don't think its a complete replacement for the current transform system until we have the following apis:

// sets the absolute local scale
set_local_scale(&mut self, scale: Vec3)

// gets the absolute local rotation
local_rotation(&self) -> Quat

// sets the absolute local rotation
set_local_rotation(&mut self, rotation: Quat)